add gmail plugin

racoonx2p opened the pull request:
Thanks for contributing the Netlify plugins directory!

**Are you adding a plugin or updating one?**

- [x] Adding a plugin
- [ ] Updating a plugin

**Have you completed the following?**

- [x] Read and followed the [plugin author guidelines](https://github.com/netlify/plugins/blob/master/docs/guidelines.md).
- [x] Included all [required fields](https://github.com/netlify/plugins/blob/master/docs/CONTRIBUTING.md#required-fields) in your entry.
- [x] Tested the plugin [locally](https://docs.netlify.com/cli/get-started/#run-builds-locally) and [on Netlify](https://docs.netlify.com/configure-builds/build-plugins/#install-a-plugin), using the plugin version stated in your entry.

**Test plan**
Link to a successful public deploy log: https://app.netlify.com/sites/netlify-plugin-gmail/deploys

**Configuration**
You need to configure this plugin using [build environment variables](https://docs.netlify.com/configure-builds/environment-variables/). How-to is included in [Github repo](https://github.com/racoonx2p/netlify-plugin-gmail#Configuration).

JGAntunes commented:
Hey @racoonx2p 👋 Thanks for your contribution! I've gone ahead and created a couple of issues under your repo to address a couple of things we've spotted.

We're still in the process of fine-tuning our [review guidelines](https://github.com/netlify/plugins/blob/master/docs/plugin_review.md). However, the functionality this plugin offers is similar to functionality we already support [natively at Netlify](https://docs.netlify.com/site-deploys/notifications/#email-notifications). Given these are not mutually exclusive though, and we can see room for your plugin offering too, we wanted to ask you if you could provide a pointer on your `README.md` to this said functionality so that folks are aware of it. For example:
```md
# netlify-plugin-gmail 📨
> A Netlify Build Plugin for Gmail
This build plugin provides an alternative to [Netlify's native e-mail notification offering](https://docs.netlify.com/site-deploys/notifications/#email-notifications) by sending e-mail from a Gmail account on every
[Netlify](https://www.netlify.com/) successful or failed build.

racoonx2p commented:
Hi @JGAntunes,
Thanks for your advice/ opened issues. I hope I addressed them all and they should be covered in the new release 1.1.0. I also updated my `README.md` with your suggestion.  
Let me know if there is something else to fix and I'll work on it! ⚒

JGAntunes commented:
Looks good @racoonx2p, thanks for putting time into this! We just need to address the conflicts in this branch by rebasing with the main branch and we should be good to go 🚀

JGAntunes commented on site/plugins.json:
```diff
@@ -422,5 +422,13 @@
+  {
+    "author": "racoonx2p",
+    "description": "Send an email from Gmail after successful/failed Netlify build.",
+    "name": "Gmail Plugin",
```
```suggestion
    "name": "Gmail",
```

We're also in the process of have consistent naming for plugins - https://github.com/netlify/plugins/issues/208 - so I believe we can drop the the `Plugin` here CC @erezrokah

JGAntunes commented:
Closing in favour of #214, rebased your work with our master branch @racoonx2p
